### DISCUSSION TITLE
NLOHMANN JSON - WRITE DATA TO A BIG FILE (500 MB)

### DISCUSSION THREAD

**Discussion by uralsib123** [nlohmann/json#3957] [Q&A]
Hi 
I have a requirement to create a big json file with  details of around 1 lakh employees in the company to a json file. 
Please suggest Nlohmann json is a good option to create big json files.  keeping json object of 1 lakh employee details in memory consumes lot of memory it seems

**Comment by nlohmann** [comment]
nlohmann/json builds an in-memory DOM tree, so holding ~100,000 employee records as a json object will cost memory roughly proportional to that structure (typically several times the raw JSON text size, due to per-value/container overhead) — there's no way around that if you build the whole thing as one json value first. For the writing side specifically, avoid j.dump() into a std::string and then writing that string to a file; instead write directly to the stream with ofs << j; (or j.dump()-to-ostream via the streaming operator<<), which streams tokens out without buffering the whole serialized text in memory. If the in-memory DOM itself is the bottleneck, consider building/writing the file incrementally in chunks (e.g. one array element/employee record at a time, writing each with `<<` and manual commas/brackets) instead of assembling one giant json object.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
